pkg/controller/apimanager/apimanager_controller.go

```diff
@@ -101,6 +105,11 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 		return err
 	}
 
+	err = c.Watch(&source.Kind{Type: &routev1.Route{}}, &handler.EnqueueRequestForObject{})
```

Comment:
As watched routes are not managed by the apimanager controller, I think it makes sense to create a separate controller for this logic. Note that this apimanager controller always tries to read APIManager type out of reconcile request object (https://github.com/3scale/3scale-operator/blob/c941a204e884f9df6eb1cccc33928db37a3ca568/pkg/controller/apimanager/apimanager_controller.go#L143)

Reply:
I will move all the Route watch, ConsoleLink creation and deletion related codes to a new Controller: webconsole_controller.go. In there, I will figure out a new way for ConsoleLink creation/deletion instead of using the existence of apimanager to determine ConsoleLink's creation/deletion.  Does this sound right to go?

Reply:
added a new controller